Barotrauma/BarotraumaShared/SharedSource/Events/Missions/GoToMission.cs

```diff
@@ -4,9 +4,21 @@ namespace Barotrauma
 {
     partial class GoToMission : Mission
     {
+        private readonly bool stateControlsCompletion;
+        private readonly int successState;
+        private readonly int displayAsFailedState;
+
         public GoToMission(MissionPrefab prefab, Location[] locations, Submarine sub)
             : base(prefab, locations, sub)
         {
+            stateControlsCompletion = prefab.ConfigElement.GetAttributeBool("statecontrolscompletion", false);
+            successState = prefab.ConfigElement.GetAttributeInt("successstate", 1);
+            displayAsFailedState = prefab.ConfigElement.GetAttributeInt("displayasfailedstate", -1);
```

Comment:
It's generally better to use `GetAttributeInt(nameof(displayAsFailedState))`, because it allows finding the reference to the field more easily, changes the argument automatically if you rename the field, and prevents doing any typos in the argument (a typo wouldn't compile).